tests/test_commands.py

```diff
@@ -262,3 +262,74 @@ def test_js_comments(self):
         cmd.process(sql)
         self.assertEqual(1, cmd._exec_and_print.call_count)
         self.assertIn("CREATE FUNCTION", cmd._exec_and_print.mock_calls[0].args[0])
+
+
+class MultipleStatementsTest(TestCase):
+
+    def test_single_line_multiple_sql_statements(self):
+        cmd = CrateShell()
+        cmd._exec_and_print = MagicMock()
+        cmd.process("SELECT 1; SELECT 2; SELECT 3;")
+        cmd._exec_and_print.assert_has_calls([
+            call("SELECT 1;"),
+            call("SELECT 2;"),
+            call("SELECT 3;"),
+        ])
+
+    def test_multiple_lines_multiple_sql_statements(self):
+        cmd = CrateShell()
+        cmd._exec_and_print = MagicMock()
+        cmd.process("SELECT 1;\nSELECT 2; SELECT 3;\nSELECT\n4;")
+        cmd._exec_and_print.assert_has_calls([
+            call("SELECT 1;"),
+            call("SELECT 2;"),
+            call("SELECT 3;"),
+            call("SELECT\n4;"),
+        ])
+
+    def test_single_sql_statement_multiple_lines(self):
+        """When processing single SQL statements, new lines are preserved."""
+
+        cmd = CrateShell()
+        cmd._exec_and_print = MagicMock()
+        cmd.process("\nSELECT\n1\nWHERE\n2\n=\n3\n;\n")
+        cmd._exec_and_print.assert_has_calls([
+            call("SELECT\n1\nWHERE\n2\n=\n3\n;"),
+        ])
+
+    def test_multiple_commands_no_sql(self):
+        cmd = CrateShell()
+        cmd._try_exec_cmd = MagicMock()
+        cmd._exec_and_print = MagicMock()
+        cmd.process("\\?\n\\connect 127.0.0.1")
+        cmd._try_exec_cmd.assert_has_calls([
+            call("?"),
+            call("connect 127.0.0.1")
+        ])
+        cmd._exec_and_print.assert_not_called()
+
+    def test_commands_and_multiple_sql_statements_interleaved(self):
+        """Combine all test cases above to be sure everything integrates well."""
+
+        cmd = CrateShell()
+        cmd._try_exec_cmd = MagicMock()
+        cmd._exec_and_print = MagicMock()
+        cmd.process("""
+    \\?
+    SELECT 1
+        WHERE 2 = 3; SELECT 4;
+    \\connect 127.0.0.1
+    SELECT 5
+        WHERE 6 = 7;
+    \\check
+        """)
+        cmd._try_exec_cmd.assert_has_calls([
```

Comment:
Maybe ordering should be somehow integrated into this testcase (eg. `\\?` gets called before `SELECT  1` statement), but I am not sure if it's worth it as it would require a [piece of code](https://stackoverflow.com/a/22677452) I do not consider elegant.